service/src/main/java/org/cbioportal/genome_nexus/service/annotation/EntrezGeneIdResolver.java

```diff
@@ -0,0 +1,82 @@
+package org.cbioportal.genome_nexus.service.annotation;
+
+import org.cbioportal.genome_nexus.model.GeneXref;
+import org.cbioportal.genome_nexus.model.TranscriptConsequence;
+import org.cbioportal.genome_nexus.model.VariantAnnotation;
+import org.cbioportal.genome_nexus.service.GeneXrefService;
+import org.cbioportal.genome_nexus.service.exception.EnsemblWebServiceException;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class EntrezGeneIdResolver
+{
+    private final CanonicalTranscriptResolver canonicalTranscriptResolver;
+    private final GeneXrefService geneXrefService;
+
+    public EntrezGeneIdResolver(CanonicalTranscriptResolver canonicalTranscriptResolver,
+                                GeneXrefService geneXrefService)
+    {
+        this.canonicalTranscriptResolver = canonicalTranscriptResolver;
+        this.geneXrefService = geneXrefService;
+    }
+
+    @Nullable
+    public String resolve(TranscriptConsequence transcriptConsequence) throws EnsemblWebServiceException
+    {
+        String entrezGeneId = null;
+
+        if (transcriptConsequence != null &&
+            transcriptConsequence.getGeneId() != null &&
+            !transcriptConsequence.getGeneId().trim().isEmpty())
+        {
+            // TODO in memory cache?
+            // check hashmap first before hitting api for gene cross-ref data
+//            if (ensemblAccessionEntrezIdMap.containsKey(transcriptConsequence.getGeneId())) {
+//                return ensemblAccessionEntrezIdMap.get(transcriptConsequence.getGeneId());
```

Comment:
This is the original code which uses an in-memory cache. Maybe we should cache the xref API call results in our database instead.